api/python/cellxgene_census/tests/experimental/diffexp/memento/test_diff_expr.py

```diff
@@ -1,156 +1,97 @@
+    in separate class.
+    """
 
-    for c, estimator in enumerate(["n_obs", "mean", "sem", "var", "selv"], start=1):
-        estimators[estimator] = range(1, len(estimators) + 1)
-        estimators[estimator] *= c
+    @pytest.fixture(scope="class", params=["test_case_1", "test_case_2"])
+    def test_cases_for_compute_all_fn(self, request: Any) -> Dict[str, Any]:
+        """
+        Fixture that generates test cases for function
+        calls to `compute_all` given a test case name.
 
-    return estimators
+        This fixture returns a tuple `(diff_exp_query, expected_result)`
+        such that `diff_exp_query` encapsulates the differential expression
+        query that will be executed by `compute_all()`.
 
+        `expected_result` is a datastructure containing the pertinent parts of
+        the return value of `compute_all()`.
+        """
+        # TODO: Figure out a common location to store estimator cube fixtures so that it is
+        # explicitly clear that both the differential expression API and differential expression cube builder
+        # components use it for testing
+        pwd = path.dirname(__file__)
+        estimator_cube_path = path.join(
+            pwd, "../../../../../../../tools/models/memento/tests/fixtures/estimators-cube-expected/"
+        )
 
-@pytest.mark.skip("Update test to work with compute_hypothesis_test()")
-@pytest.mark.parametrize("dim_counts", [{"feature_id": 3, "cell_type_ontology_term_id": 3}])
-def test__setup__too_many_treatment_values__fails(estimators_df: pd.DataFrame) -> None:
-    with pytest.raises(AssertionError, match="treatment must have exactly 2 distinct values"):
-        pass
-        # diff_expr.compute_hypothesis_test( ... )
+        test_cases = {
+            "test_case_1": {
+                "diff_exp_query": {
+                    "cube_path": estimator_cube_path,
+                    "query_filter": "tissue_general_ontology_term_id in ['UBERON:0001723'] and sex_ontology_term_id in ['PATO:0000383', 'PATO:0000384']",
+                    "treatment": "sex_ontology_term_id",
+                    "num_sampled_genes": 2,
+                },
+                "expected_diff_exp_result": [
+                    ("ENSG00000000419", -0.111612, -1.895204, 0.058065),
+                    ("ENSG00000002330", 0.229054, 4.085651, 0.000044),
+                ],
+            },
+            "test_case_2": {
+                "diff_exp_query": {
+                    "cube_path": estimator_cube_path,
+                    "query_filter": "tissue_general_ontology_term_id in ['UBERON:0001723'] and cell_type_ontology_term_id in ['CL:0000066', 'CL:0000057']",
+                    "treatment": "cell_type_ontology_term_id",
+                    "num_sampled_genes": 2,
+                },
+                "expected_diff_exp_result": [
+                    ("ENSG00000000419", 0.868715, 6.048411, 1.462810e-09),
+                    ("ENSG00000002330", 0.834346, 5.218739, 1.801458e-07),
```

Comment:
Unless we're confident that these values are "correct", I would add an issue and corresponding comment to indicate the values require manual validation (by a comp bio person). I'm assuming we have plans to do another level of biological validation. Once the implementation is shown to be producing valid results, we should link out to the outcomes of the manual validation effort (a notebook, etc.) as "proof" of these data's correctness.

Reply:
Thanks @atolopko-czi - Does this capture your suggestion? https://github.com/chanzuckerberg/cellxgene-census/issues/977

Reply:
yup!